### PR TITLE
Relate frontend query and background session query

### PR DESCRIPTION
**Pull request by xzxiong**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #13384

## What this PR does / why we need it:
changes:
1. reset backgroud session id as frontend query's statement id.
2. reset fronted query's txn-id if the txn-id is non-setted, by using background session txn-id
    - default skip the background transaction, which DO check privileges. 
3. fix stmt reset while do free
4. improve statement aggr logic, if all aggr stmts in the same txn, not reset the txn-id. @gavinyue

### CODE REVIEW

File: pkg/frontend/authenticate.go [hunk 1]
```diff
@@ -7734,6 +7729,13 @@ func InitGeneralTenant(ctx context.Context, ses *Session, ca *tree.CreateAccount
 			return rtnErr
 		}
 
+		//USE the mo_catalog
```
Comment (xzxiong) [inline pkg/frontend/authenticate.go:7732]: @daviszhen 看下这里，我挪了下位置。为了保证仅 “一个事务”。

File: pkg/frontend/mysql_cmd_executor.go [hunk 2]
```diff
@@ -371,6 +371,29 @@ var RecordStatementTxnID = func(ctx context.Context, ses *Session) {
 		}
 		stm.Report(ctx)
 	}
+
+	// set frontend statement's txn-id
+	if upSes := ses.upstream; upSes != nil && upSes.tStmt != nil && upSes.tStmt.IsZeroTxnID() /* not record txn-id */ {
+		// background session has valid txn
+		if handler := ses.GetTxnHandler(); handler.IsValidTxnOperator() {
+			_, txn, err = handler.GetTxn()
```
Comment (daviszhen) [inline pkg/frontend/mysql_cmd_executor.go:379]: 如果上面isValidOperator为true，用GetTxnOperator接口。

File: pkg/util/trace/impl/motrace/report_statement.go [hunk 3]
```diff
@@ -310,24 +320,42 @@ func (s *StatementInfo) freeNoLocked() {
...
+	s.Account = ""
+	s.User = ""
+	s.Host = ""
 	s.RoleId = 0
+	s.Database = ""
 	s.Statement = ""
+	s.StmtBuilder.Reset()
 	s.StatementFingerprint = ""
 	s.StatementTag = ""
-	s.FreeExecPlan()
```
Comment (daviszhen) [inline pkg/util/trace/impl/motrace/report_statement.go:317]: 确定要删吗?

**Review by daviszhen — CHANGES_REQUESTED** [review]

File: pkg/util/trace/impl/motrace/report_statement.go (hunk 3 above)
Comment (xzxiong) [reply pkg/util/trace/impl/motrace/report_statement.go:317]: statement_info 对 ExecPlan是只读的，execPlan的生命周期是由 frontend 管理。

Comment (xzxiong) [reply pkg/util/trace/impl/motrace/report_statement.go:317]: 另外，这个调用并没有去掉。代码移动了 [L343](https://github.com/matrixorigin/matrixone/pull/13436/files#diff-6bb6c3e48a06b6a752b1a028ea13f3f2973f20039dcbce2759e65cf30e50da2aR343)

**Review by daviszhen — APPROVED** [review]

**Pull request merged**